raft/raft.go

```diff
@@ -441,30 +441,37 @@ func (r *raft) getProgress(id uint64) *Progress {
 	return r.learnerPrs[id]
 }
 
-// sendAppend sends RPC, with entries to the given peer.
-func (r *raft) sendAppend(to uint64) {
+// sendAppend sends an append RPC with new entries to the given peer,
+// if necessary. Returns true if a message was sent. The sendIfEmpty
+// argument controls whether messages with no entries will be sent
+// ("empty" messages are useful to convey updated Commit indexes, but
+// are undesirable when we're sending multiple messages in a batch).
+func (r *raft) sendAppend(to uint64, sendIfEmpty bool) bool {
```

Comment:
can we keep the old sendAppend? and add a new func maybeSendAppend (or sendAppendIfNotEmpty)?